main/ui/src/main/java/org/cryptomator/ui/Main.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2014 Sebastian Stenzel
+ * This file is licensed under the terms of the MIT license.
+ * See the LICENSE.txt file for more info.
+ * 
+ * Contributors:
+ *     Sebastian Stenzel - initial API and implementation
+ ******************************************************************************/
+package org.cryptomator.ui;
+
+import java.io.File;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+import javafx.application.Application;
+
+import org.controlsfx.tools.Platform;
+import org.cryptomator.ui.util.SingleInstanceManager;
+import org.cryptomator.ui.util.SingleInstanceManager.RemoteInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.axet.desktop.os.mac.AppleHandlers;
+
+public class Main {
+	public static final Logger LOG = LoggerFactory.getLogger(MainApplication.class);
+
+	public static final CompletableFuture<Consumer<File>> openFileHandler = new CompletableFuture<>();
+
+	public static void main(String[] args) {
+		if (Platform.getCurrent().equals(org.controlsfx.tools.Platform.OSX)) {
+			/*
+			 * On OSX we're in an awkward position. We need to register a
+			 * handler in the main thread of this application. However, we can't
+			 * even pass objects to the application, so we're forced to use a
+			 * static CompletableFuture for the handler, which actually opens
+			 * the file in the application.
+			 */
+			try {
+				AppleHandlers.getAppleHandlers().addOpenFileListener(list -> {
+					try {
+						openFileHandler.get().accept(list);
+					} catch (Exception e) {
+						LOG.error("exception handling file open event", e);
+						throw new RuntimeException(e);
+					}
+				});
+			} catch (Throwable e) {
```

Comment:
No exceptions declared in addOpenFileListener -> remove catch block or switch to specific RuntimeException, if one is expected here.